services/swarm/client/client.go

```diff
@@ -0,0 +1,204 @@
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"sync/atomic"
+)
+
+type Config struct {
+	URLs       []string
+	APIVersion string
+	TLSConfig  *tls.Config
+}
+
+type Client interface {
+	Scales(name string) ScalesInterface
+	Update(c Config) error
+}
```

Comment:
Hmm so this makes me wonder if we should create a common autoscale node implementation and the provide the swarm vs k8s implementation? If the client interface is really the same then maybe we should, thoughts?

To be clear I am not suggesting we merge the pipline (i.e. TICKscript) nodes, but rather just the executing node implementation to have a common backend. So TICKscript will remain the same as you have implemented it but the `kapacitor/k8s_autoscale.go` and `kapacitor/swarm_autoscale.go` nodes would share a common node implementation.

Since you were in the code most recently what do you think? Does it make sense to merge the backends but keep the TICKscript interface separate?